doc/conventions.md

```diff
@@ -0,0 +1,92 @@
+# Conventions
+
+- [State](#state)
+- [Monads](#monads)
+- [Action](#action)
+- [Handler](#handler)
+- [Dependencies](#dependencies)
+- [Interceptors](#interceptors)
+
+## State
+
+A state record. It is created for each HTTP request and represents the current state of the application. It contains:
+
+- the application's dependencies
+- request
+- request-data
+- response
+
+This structure is very volatile, will be updated quite often on the application's life cycle.
+
+The main modules that updates the state are:
+
+- Routes:
+
+  Add information from the matched route to the state map
+
+- Interceptors:
+
+  Add, consumes or remove information from the state map. More details in [Interceptors](#interceptors) section.
+
+- Actions:
+
+  In actions, you are able to interfere with the :leave parts of the interceptors.
+
+At the last step of execution the handler extracts the response value from the state.
+
+The state is renewed on every request.
+
+## Monads
+
+"Monad is a simple and powerful mechanism for function composition that helps us to solve very common IT problems such
```

Comment:
I think this should be "The monad" or "Monads are". I know it's a quote, I suspect the original is wrong too.